[Bug]: ToolResponse can downgrade an error state when appending later chunks

## Prerequisites

- [x] I searched existing issues, discussions, and open/closed pull requests; I did not find a duplicate.
- [x] This is a bug rather than a usage question.

## Background / Description

`ToolResponse.append_chunk()` documents that it preserves the failure state and keeps the previous state when a later chunk is not worse. However, after an `ERROR` chunk, a later `INTERRUPTED` or `DENIED` chunk overwrites the aggregate response state.

A streamed tool response can therefore finish with a non-error state even though one of its chunks failed. The aggregate should keep `ToolResultState.ERROR` once observed.

## Error Messages

No exception is raised. The incorrect aggregate state is:

```text
ToolResultState.INTERRUPTED
```

(or `ToolResultState.DENIED`) instead of `ToolResultState.ERROR`.

## Steps to Reproduce

```python
from agentscope.message import TextBlock, ToolResultState
from agentscope.tool import ToolChunk, ToolResponse

response = ToolResponse()
response.append_chunk(
    ToolChunk(
        content=[TextBlock(text=failed)],
        state=ToolResultState.ERROR,
    ),
)
response.append_chunk(
    ToolChunk(
        content=[TextBlock(text=interrupted)],
        state=ToolResultState.INTERRUPTED,
    ),
)

assert response.state == ToolResultState.ERROR
```

The assertion currently fails because `response.state` is `ToolResultState.INTERRUPTED`.

## Environment

- AgentScope Version: 2.0.5 (current `main`, commit `999f64d`)
- Python Version: 3.12.13
- OS: Windows 11

Reproduced at HEAD `27b6a0d2` (Python 3.12.13, clean `python:3.12-slim` container, `pip install -e .`):

```
state after ERROR then INTERRUPTED: interrupted
AssertionError: expected ERROR, got interrupted
```

Also ran `DENIED` -> `INTERRUPTED`, same downgrade: state ends as `interrupted`.

Root cause is `ToolResponse.append_chunk` in `src/agentscope/tool/_response.py:136-141`:

```python
if chunk.state == ToolResultState.ERROR:
    self.state = ToolResultState.ERROR
elif chunk.state == "interrupted":
    self.state = ToolResultState.INTERRUPTED
elif chunk.state == ToolResultState.DENIED:
    self.state = ToolResultState.DENIED
```

The comment above it ("Only reserve the failure state and keep the previous state if not worse") describes a priority ordering, but the code is a plain if/elif on the incoming chunk's state alone, with no check against `self.state`'s current value. Whichever branch the new chunk matches wins outright, so any later chunk downgrades an earlier `ERROR` or `DENIED` to whatever state matches next.

A precedence check before assigning would fix it, e.g. only overwrite when the new state is not already less severe than the current `self.state`.

--- a/src/agentscope/tool/_response.py
+++ b/src/agentscope/tool/_response.py
@@ -135,9 +135,15 @@ class ToolResponse(BaseModel):
         # worse.
         if chunk.state == ToolResultState.ERROR:
             self.state = ToolResultState.ERROR
-        elif chunk.state == "interrupted":
+        elif (
+            self.state != ToolResultState.ERROR
+            and chunk.state == ToolResultState.INTERRUPTED
+        ):
             self.state = ToolResultState.INTERRUPTED
-        elif chunk.state == ToolResultState.DENIED:
+        elif (
+            self.state != ToolResultState.ERROR
+            and chunk.state == ToolResultState.DENIED
+        ):
             self.state = ToolResultState.DENIED
 
         self.metadata.update(chunk.metadata)

